service/controller/resource/capzcrs/create.go

```diff
@@ -38,6 +39,23 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
+	{
+		objKey := client.ObjectKey{
+			Namespace: cr.Namespace,
+			Name:      cr.Name,
+		}
+		cluster := new(capiv1alpha3.Cluster)
+		err = r.ctrlClient.Get(ctx, objKey, cluster)
+		if errors.IsNotFound(err) {
+			// all good
+		} else if err != nil {
+			return microerror.Mask(err)
+		} else if cluster.GetDeletionTimestamp() != nil {
```

Comment:
I think we usually use the IsZero() function on timestamps

Reply:
Already fixed by Jose. Go team!